spras/interactome.py

```diff
@@ -0,0 +1,123 @@
+    - with bi-directed edges, we are not loosing too much information because the relationship of the undirected egde is still preserved
+
+   * A user must keep the Direction column when using this function
+
+    @param df: input network df of edges, weights, and directionality
+    @return a dataframe with no undirected edges in Direction column
+    """
+
+    mask = df['Direction'] == 'U'
+    new_df = df[mask].copy(deep=True)
+    new_df['Interactor1'], new_df['Interactor2'] = new_df['Interactor2'], new_df['Interactor1']
+    new_df['Direction'] = 'D'
+    df.loc[mask, 'Direction'] = 'D'
+    df = pd.concat([df, new_df], ignore_index=True)
+    return df
+
+
+def convert_directed_to_undirected(df: pd.DataFrame) -> pd.DateOffset:
+    """
+    turns a graph into a fully undirected graph
+    - turns all the directed edges directly into undirected edges
+    - we will loose any sense of directionality and the graph won't be inherently accurate, but the basic relationship between the two connected nodes will still remain intact.
+
+    * A user must keep the Direction column when using this function
+
+    @param df: input network df of edges, weights, and directionality
+    @return a dataframe with no directed edges in Direction column
+    """
+
+    df["Direction"] = "U"
+
+    return df
+
+
+def add_constant(df: pd.DataFrame, new_col_name: str, const: str) -> pd.DataFrame:
+    """
+    adds a constant at the end of the input dataframe that is needed inbetween columns for a specifc algorithm
+
+    @param df: input network df of edges, weights, and directionality
+    @param new_col_name: the name of the new column
+    @param const: some type of constant needed in the df
+    @return a df with a new constant added to every row
+    """
+
+    df.insert(df.shape[1], new_col_name, const)
+
+    return df
+
+
+def add_directionality_constant(df: pd.DataFrame,  col_name: str, dir_const: str, undir_const: str) -> pd.DataFrame:
+    """
+    deals with adding in directionality constants for mixed graphs that aren't using the universal input directly
+
+    * user must keep the Direction column when using the function
+
+    @param df: input network df of edges, weights, and directionality
+    @param col_name: the name of the new column
+    @param dir_const: the directed edge sep
+    @param undir_const: the undirected edge sep
+    @return a df converted to show directionality differently
+    """
+    df.insert(df.shape[1], col_name, dir_const)
+    mask = df['Direction'] == 'U'
+    df.loc[mask, col_name] = undir_const
+
+    return df
+
+def readd_direction_col_mixed(df: pd.DataFrame, existing_direction_column: str, dir_const: str, undir_const: str) -> pd.DataFrame:
+    """
+    readds a 'Direction' column that puts a 'U' or 'D' at the end of provided dataframe
+    based on the dir/undir constants in the existing direction column
+
+    *user must keep the existing direction column when using the function
```

Comment:
I'm still unsure what these comments are suggesting. Does this mean don't delete the Direction column after calling this function?